dirsrvtests/tests/suites/acl/aci_excl_filter_test.py

```diff
@@ -0,0 +1,146 @@
+           - Search for user(s) ' admin in the subtree that satisfy this criteria:
+               DEFAULT_SUFFIX, ldap.SCOPE_SUBTREE, cn_filter, [deniedattr, 'dn'] 
+        6.  The search should return 2 entries with the username 'admin'
+        7.  Verify that the users found do not have the --> deniedattr = 'telephonenumber' marker
+    :expectedresults:
+        1. Operation should be successful
+        2. Operation should be successful
+        3. Operation should be successful
+        4. PASS - users found do not have the --> deniedattr = 'telephonenumber' marker
+
+    """
+    log.info('Bind as root DN')
+  
+    log.info('Create an OU for them')    
+    ous = OrganizationalUnit(topo.standalone, DEFAULT_SUFFIX)
+#   # Create the OU for them
+  
+    log.info('Create an top org users')
+    users = UserAccounts(topo.standalone, DEFAULT_SUFFIX)
+    log.info('Add aci which contains extensible filter.')
+    ouname = 'outest'
+    username = 'admin'
+    passwd = 'Password'
+    deniedattr = 'telephonenumber'
+    log.info('Add aci which contains extensible filter.')
+
+    aci_text = ('(targetattr = "{}")'.format(deniedattr) +
+                 '(target = "ldap:///{}")'.format(DEFAULT_SUFFIX) +
+                 '(version 3.0;acl "admin-tel-matching-rule-outest";deny (all)' +
+                 '(userdn = "ldap:///{}??sub?(&(cn={})(ou:dn:={}))");)'.format(DEFAULT_SUFFIX, username, ouname))
+    try:
+         topo.standalone.modify_s(DEFAULT_SUFFIX, [(ldap.MOD_ADD, 'aci', ensure_bytes(aci_text))])
+    except ldap.LDAPError as e:
+         log.error('Failed to add aci: (%s) error %s' % (aci_text, e.args[0]['desc']))
+         assert False
+
+    log.info('Add entries ...')
+    for idx in range(0, 2):
+        ou0 = 'OU%d' % idx
+        log.info('Adding "ou" : %s under "dn" : %s...' % (ou0, DEFAULT_SUFFIX))
+        add_ou_entry(topo, ou0, DEFAULT_SUFFIX)
+        
+        parent = 'ou=%s,%s' % (ou0, DEFAULT_SUFFIX)
+        log.info('Adding %s under %s...' % (ouname, parent))
+        add_ou_entry(topo, ouname, parent)
+    
+    user = UserAccounts(topo.standalone, parent, rdn=None)
+    for idx in range(0, 2):
+        parent = 'ou=%s,ou=OU%d,%s' % (ouname, idx, DEFAULT_SUFFIX)
+        username = '{}{}'.format(username, idx)
+        log.info('Adding User: %s under %s...' % (username, parent))
+        
+        user = add_user_entry(topo, user, username, passwd, parent)
+
+
+    # binddn = 'cn=%s,%s' % (username, parent)
+    log.info('Bind as user %s' % username)
+    binddn_user = user.get(username)
+    
+    try:
+        conn = binddn_user.bind(passwd)
+        
+    except :
+        log.error(" {} failed to authenticate: ".format(binddn_user))
+        assert False
+
+    cn_filter = '(cn=%s)' % username
+    try:
+        # entries = topo.standalone.search_s(DEFAULT_SUFFIX, ldap.SCOPE_SUBTREE, cn_filter, [deniedattr, 'dn'])
+        entries = user.get((DEFAULT_SUFFIX, cn_filter, [deniedattr, 'dn']))
```

Comment:
I am getting a failure here with entries...I am expecting to get a list of length 2

Reply:
Resolved